src/editor/Collaboration.js

```diff
@@ -0,0 +1,109 @@
+define(function (require, exports, module) {
+    "use strict";
+
+    var SimpleWebRTC    = require("simplewebrtc");
+
+    function Collaboration() {
+        var webrtc = new SimpleWebRTC({
+            // the id/element dom element that will hold "our" videos
+            localVideoEl: 'localVideo',
+            // the id/element dom element that will hold remote videos
+            remoteVideosEl: 'remotesVideos',
+            // immediately ask for camera access
+            autoRequestMedia: false
+        });
+        this.webrtc = webrtc;
+        this.pending = []; // pending clients that need to be initialized.
+        this.changing = false;
+    };
+
+    Collaboration.prototype.init = function(codemirror) {
+        var self = this;
+        this.webrtc.joinRoom('thimble', function() {
+            self.codemirror = codemirror;
+            self.webrtc.sendToAll("new client", {});
+            self.addListeners();
+        });
+    };
+
+    Collaboration.prototype.addListeners = function() {
+        var self = this;
+        this.webrtc.on("createdPeer", function(peer) {
+            self.initialiseNewClient(peer);
+        });
+
+        this.webrtc.connection.on('message', function (msg) {
+            self.handleMessage(msg);
+        });
+    };
+
+    Collaboration.prototype.handleMessage = function(msg) {
+        if(msg.type === "new client") {
+            this.addToPending(msg.from);
+        }
+        if(this.changing) {
+            return;
+        }
+        if(msg.type === "data") {
+            this.handleCodemirrorChange(msg.payload);
+        }
+        if(msg.type === "initClient") {
+            this.initiseEditor(msg.payload);
+        }
+    };
+
+    Collaboration.prototype.addToPending = function(id) {
+        this.pending.push(id);
+    };
+
+    Collaboration.prototype.initiseEditor = function(value) {
+        this.changing = true;
+        this.codemirror.setValue(value);
```

Comment:
Move this inline above to `handleMessage` since it's so small, and never called again.